src/redisjson.rs

```diff
@@ -333,14 +333,18 @@ impl RedisJSON {
         Ok(res.into())
     }
 
-    // TODO: Rename this to 'get_value', since 'doc' is overloaded.
-    pub fn get_doc<'a>(&'a self, path: &'a str) -> Result<&'a Value, Error> {
-        let results = jsonpath_lib::select(&self.data, path)?;
+    pub fn get_first<'a>(&'a self, path: &'a str) -> Result<&'a Value, Error> {
+        let results = self.get_values(path)?;
         match results.first() {
             Some(s) => Ok(s),
             None => Err("ERR path does not exist".into()),
         }
     }
+
```

Comment:
Instead of separate `get_values` and `get_first` methods, I suggest implementing `Iterator` (similar to what I did with `redisearch_api::ResultsIterator`): Replace `get_values` with a method named `iter_values` that will return an `Iterator<Item=Value>`. Of course, the underlying API doesn't return an Iterator but a `Vec` so this will not improve performance, but it's a cleaner API (and if the underlying implementation changes this will be transparent).

Then, the standard `first` method can be used on the resulting iterator (note that we already call `first` on the results in `commands::create_document` and other places).

The common pattern of requiring at least one result and returning an error otherwise can be accomplished in the standard way by calling `first().unwrap_or(...)`.